pkg/stardogapi/permissions.go

```diff
@@ -0,0 +1,43 @@
+package stardogapi
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+type getRolePermissionsRolesResponse struct {
+	Permissions []Permission `json:"permissions"`
+}
+
+// Get Permissions of a role
+func (c *Client) GetRolePermissions(ctx context.Context, name string) ([]Permission, error) {
+	var response getRolePermissionsRolesResponse
+
+	return response.Permissions, c.sendRequest(ctx,
+		http.MethodGet,
+		fmt.Sprintf("/admin/permissions/role/%s", name),
+		nil,
+		&response,
+	)
+}
+
+// Add the Permission to a role
+func (c *Client) AddRolePermission(ctx context.Context, name string, permission Permission) (err error) {
+	return c.sendRequest(ctx,
+		http.MethodPut,
+		fmt.Sprintf("/admin/permissions/role/%s", name),
```

Comment:
This might be dangerous if people can modify role names in any way to call other admin endpoints (`../`). I'd feel safer if name is sanitized.

Reply:
Added path value sanitizing 👍